helpdesk-ai/src/main/java/org/symphonyoss/symphony/bots/ai/message/MessageProducer.java

```diff
@@ -60,10 +77,29 @@ private SymMessage buildMessage(SymphonyAiMessage symphonyAiMessage) {
       }
     }
 
-    message.append(symphonyAiMessage.getAiMessage());
+    if (symphonyAiMessage.getMessageData() != null) {
+      content.append(this.parseMessage(symphonyAiMessage.getMessageData()));
+      message.append(content);
+    } else {
+      message.append(symphonyAiMessage.getAiMessage());
+    }
+
     message.append("</messageML>");
 
     SymMessage symMessage = new SymMessage();
+
+    if (userId != null && hasAttachment(symphonyAiMessage)) {
+      SymMessage attachmentMessage = new SymMessage();
+      attachmentMessage.setId(symphonyAiMessage.getMessageId());
+      attachmentMessage.setStreamId(symphonyAiMessage.getStreamId());
+
+      List<SymAttachmentInfo> attachmentInfoList = symphonyAiMessage.getAttachments();
+
+      File file =
```

Comment:
Why do you only get the first attachment?

Reply:
oops my bad, test purposes only that I later forgot to refactor